### PR TITLE
Feat 618 - [Doc] IAM roles for pods or service level for apps

### PR DESCRIPTION
**Pull request by yuriadams**
[task 618](https://github.com/convox/issues-private/issues/618)

### CODE REVIEW

File: docs/reference/primitives/app/service.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ services:
 | Attribute     | Type       | Default             | Description                                                                                                                                |
 | ------------- | ---------- | ------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
 | **agent**       | boolean    | false               | Set to **true** to declare this Service as an [Agent](/configuration/agents)                                                      |
-| **annotations** | list       |                     | A list of annotation keys and values to populate the metadata for the deployed pods and their serviceaccounts                              |
+| **annotations*** | list       |                     | A list of annotation keys and values to populate the metadata for the deployed pods and their serviceaccounts                              |
```
Comment (nightfury1204) [inline docs/reference/primitives/app/service.md:76]: `**annotations***` extra start is that intentional or mistake?

File: docs/reference/primitives/app/service.md [hunk 2]
```diff
@@ -97,6 +97,26 @@ services:
 
 > Environment variables declared on `convox.yml` will be populated for a Service.
 
+#### *annotations
```
Comment (nightfury1204) [inline docs/reference/primitives/app/service.md:100]: start `*` before the annotation name is that intentional or mistake?

**Review by ntner — APPROVED** [review]
looks good to me

File: docs/reference/primitives/app/service.md [hunk 3]
```diff
@@ -97,6 +97,26 @@ services:
...
+You can use annotations to attach arbitrary non-identifying metadata to objects. Clients such as tools and libraries can retrieve this metadata. In convox, annotations will reflect in pods and service accounts.
```
Comment (Twsouza) [inline docs/reference/primitives/app/service.md:101]: nit pick

```
On Convox, 
```

File: docs/reference/primitives/app/service.md [hunk 4]
```diff
@@ -97,6 +97,26 @@ services:
...
+
+Here are some examples of information that could be recorded in annotations:
```
Comment (Twsouza) [inline docs/reference/primitives/app/service.md:103]:
```
information that can be recorded
```

File: docs/reference/primitives/app/service.md [hunk 5]
```diff
@@ -97,6 +97,26 @@ services:
...
+- Build, release, or image information like timestamps, release IDs, git branch, PR numbers, image hashes, and registry address.
+- Fields managed by a declarative configuration layer. Attaching these fields as annotations distinguishes them from default values set by clients or servers, and from auto-generated fields and fields set by auto-sizing or auto-scaling systems.
+- User or tool/system provenance information, such as URLs of related objects from other ecosystem components.
+- Configure a service to assume an IAM Role(aws and gcp only). For example:
```
Comment (Twsouza) [inline docs/reference/primitives/app/service.md:107]: Use capitals

```
AWS and GCP only
```

File: docs/reference/primitives/app/service.md [hunk 6]
```diff
@@ -97,6 +97,26 @@ services:
 
 > Environment variables declared on `convox.yml` will be populated for a Service.
 
+#### *annotations
+You can use annotations to attach arbitrary non-identifying metadata to objects. Clients such as tools and libraries can retrieve this metadata. On convox, annotations will reflect in pods and service accounts.
```
Comment (Twsouza) [inline docs/reference/primitives/app/service.md:101]: Convox with capital C

```
On Convox
```

**Review by Twsouza — APPROVED** [review]

**Pull request merged**